### PR TITLE
fix: various task transformer improvements

### PR DESCRIPTION
**Pull request by degenaro**
for xlsx
- remove support for column "Version"
- add support for columns "goal_version", "rule_name_id", "rule_version"
- add support for profile-type generation "by-goal", "by-rule",
"by-control", "by-check"
- skip rows where control_id or goal_id is missing with complaint
- skip rows where control_id and goal_id are missing, after 100
consecutive terminate processing
- improve sotring of controls

for ocp4
- give warning (not failure) when unable to extract title from
compliance-as-code rule.yml

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

Fixes fixes and improvements to handle revised spread sheet format, plus small fix for OCP4 to lessen severity of missing title.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: tests/trestle/tasks/xlsx_to_oscal_profile_test.py [hunk 1]
```diff
@@ -81,6 +81,86 @@ def test_xlsx_execute(tmp_path):
...
+    assert retval == TaskOutcome.SUCCESS
+    list_dir = os.listdir(d_produced)
+    assert len(list_dir) == 1
+    assert d_expected != d_produced
+    for fn in list_dir:
+        f_expected = d_expected / fn
+        f_produced = d_produced / fn
+        result = text_files_equal(f_expected, f_produced)
+        assert (result)
+
```
Comment (fsuits) [inline tests/trestle/tasks/xlsx_to_oscal_profile_test.py:108]: I don't think there is a need for parens on result - and it looks like you do it for most of your tests.  Ideally they would all be changed simply to `assert result` but I guess it isn't critical.

File: trestle/tasks/xlsx_to_oscal_profile.py [hunk 2]
```diff
@@ -123,20 +126,90 @@ def _execute(self) -> TaskOutcome:
         self._report_issues()
         return TaskOutcome('success')
 
-    def _get_include_controls(self) -> List[SelectControlById]:
-        """Get include controls from spread sheet."""
-        include_control = SelectControlById(with_ids=self._get_with_ids())
-        return [include_control]
+    def _get_imports_by_goal(self) -> List[Import]:
+        """Get imports by goal."""
+        return self._get_imports_by_check()
```
Comment (fsuits) [inline trestle/tasks/xlsx_to_oscal_profile.py:131]: Do you really need this other entry for get_imports_by_check()?  Maybe it makes the code more clear.

File: trestle/tasks/xlsx_to_oscal_profile.py [hunk 3]
```diff
@@ -123,20 +126,90 @@ def _execute(self) -> TaskOutcome:
...
+        if '.' in control:
+            k3 = int(control.split('-')[1].split('.')[1])
+        else:
+            k3 = 0
+        return (k1, k2, k3)
+
+    def _oscal_namify(self, control: str) -> str:
+        """Rectify parenthesized numbers in controls."""
+        control = control.replace('(', '.')
+        control = control.replace(')', '')
```
Comment (fsuits) [inline trestle/tasks/xlsx_to_oscal_profile.py:211]: Can simplify by chaining the two replace calls into one line.

**Review by fsuits — APPROVED** [review]
LGTM - a few minor comments as feedback but not critical

File: tests/trestle/tasks/xlsx_to_oscal_profile_test.py (hunk 1 above)
Comment (degenaro) [reply tests/trestle/tasks/xlsx_to_oscal_profile_test.py:108]: I opened #1101 and will fix all with a subsequent PR sometime after trestle 1.0.0.

File: trestle/tasks/xlsx_to_oscal_profile.py (hunk 2 above)
Comment (degenaro) [reply trestle/tasks/xlsx_to_oscal_profile.py:137]: I think this is fine for now. It is a way to document that `by_goal` and `by_check` mean the same thing.

File: trestle/tasks/xlsx_to_oscal_profile.py (hunk 3 above)
Comment (degenaro) [reply trestle/tasks/xlsx_to_oscal_profile.py:217]: I dunno, for one less line my eye likes the code as is..

**Comment by degenaro** [comment]
I made one last change after consultation with Anca. In the case of `by-control`, we don't presently have the correct parameters in the spread sheet, so the emitted parameters in the OSCAL Profile are excess baggage and have therefore been removed. 

This is now good to go.

**Pull request merged**